src/commands/ui.rs

```diff
@@ -10,37 +10,66 @@ use ring::{aead, agreement, digest, rand, signature};
 use crate::commands::ExitCode;
 use crate::options::{Options, UI};
 use crate::platform::data_dir;
-use crate::portable::local::{instance_data_dir, InstanceInfo};
+use crate::portable::local::{instance_data_dir, NonLocalInstance};
 use crate::print;
 
 pub fn show_ui(options: &Options, args: &UI) -> anyhow::Result<()> {
     let connector = options.create_connector()?;
     let builder = connector.get()?;
     let mut url = format!("http://{}:{}/ui", builder.get_host(), builder.get_port());
-    if let Some(instance) = builder.get_instance_name() {
-        if instance != "_localdev" {
-            let ver = InstanceInfo::read(instance)?.get_version()?.specific();
-            if ver.major < 2 {
+
+    if !args.no_server_check {
+        match open_url(&url).map(|r| r.status()) {
+            Ok(reqwest::StatusCode::OK) => (),
+            Ok(reqwest::StatusCode::NOT_FOUND) => {
+                print::error("the specified EdgeDB server is not serving Web UI.");
+                print::echo!(
+                    "  If you have EdgeDB 2.0 and above, try to run the \
+                    server with `--admin-ui=enabled`."
+                );
+                return Err(ExitCode::new(2).into());
+            }
+            Ok(_) => {
```

Comment:
Why are you hiding status code here? Print it at least in `log::info`.

Reply:
Also, we're accessing without token here. Does EdgeDB return 200 in such a case? I think it should return 403 or something.

Reply:
Right I thought it was returning 401 too, but this is just the `index.html` that is either 200 or 404.